start-mongodb.sh

```diff
@@ -1,3 +1,18 @@
 #!/bin/sh
 
-docker run --name mongodb --publish 27017:27017 --detach mongo:$1
+REPL_COMMAND=""
+
+if [ -n "$MONGO_REPL_SET" ]
+then
+    REPL_COMMAND="--bind_ip_all --replSet $MONGO_REPL_SET"
+fi
+
+docker run --name mongodb --publish 27017:27017 --detach mongo:$1 $REPL_COMMAND
+
+if [ -n "$MONGO_REPL_SET" ]
+then
+    echo "initiating $MONGO_REPL_SET."
+    sleep 1
+    docker exec -t mongodb mongo --eval "rs.initiate({ \"_id\": \"$MONGO_REPL_SET\", \"members\": [{ \"_id\": 0, \"host\": \"localhost\" }] })"
```

Comment:
The secret sauce.

I was able to connect to the dockerized when I didn't create a replica set which pointed to the fact that the replica set initialization did some network stuff. Indeed the `rs.initiate()` command seems to bind an entry point to the replica set on a certain hostname which is the container id (`ef2bdf8b4079`)

```
$ mongo --eval "rs.status()"
...
members" : [
                {
                        "_id" : 0,
                        "name" : "ef2bdf8b4079:27017",
...

$ docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                      NAMES
ef2bdf8b4079        mongo:4.2           "docker-entrypoint.s…"   22 seconds ago      Up 20 seconds       0.0.0.0:27017->27017/tcp   mongodb

Reply:
Yes, good catch! Using `host: 'localhost'` fixes the networking issue! Thank you